### PR TITLE
1.x: fix SerializedObserverTest.testNotificationDelay

### PR DESCRIPTION
**Pull request by akarnokd**
This PR fixes the `SerializedObserverTest.testNotificationDelay` test to use reentrancy to verify the buffer in `SerializedObserver` is properly drained. For now, it loops 10,000 times to see if the CI brings up the issue again. Takes 400ms on my machine but I'll probably reduce it to 1,000 times.

(The previous version was a bit convoluted with all those countdowns that didn't properly lock out the other thread).

Reported in issue #1389

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
:+1: 

// Took me awhile to understand what are you doing in the test

File: src/test/java/rx/observers/SerializedObserverTest.java [hunk 1]
```diff
@@ -255,74 +257,62 @@ public void runConcurrencyTest() {
...
-                TestSubscriber<String> to = new TestSubscriber<String>(new Observer<String>() {
-
+                
+                @SuppressWarnings("unchecked")
+                final Observer<Integer>[] os = new Observer[1];
+                
+                final List<Thread> threads = new ArrayList<Thread>();
+                
+                final Observer<Integer> o = new SerializedObserver<Integer>(new Observer<Integer>() {
+                    boolean first;
```
Comment (stevegury) [inline src/test/java/rx/observers/SerializedObserverTest.java]: Don't you think that being explicit improve readability?
`boolean first = false;`

**Comment by stevegury** [comment]
:+1:

File: src/test/java/rx/observers/SerializedObserverTest.java (hunk 1 above)
Comment (akarnokd) [inline src/test/java/rx/observers/SerializedObserverTest.java]: No, there is no proper guarantee in the JVM that default-valued initializers get erased and I personally don't like it either.

**Pull request merged**